File: Cpp/Creational_Singleton/Singleton/goverment.cpp
#include "goverment.h"

Goverment* Goverment::instance = NULL;

Goverment::Goverment()
{
    cout << "Constructor!" <<endl;
}

Goverment *Goverment::getInstance()
{
    if(Goverment::instance == NULL)
        Goverment::instance = new Goverment();
    return Goverment::instance;
}

File: CSharp/Creational_Singleton/Creational_Singleton/Goverment.cs
using System;
namespace Creational_Singleton
{
	public class Goverment
	{
		// Our instance, create only once
		private static readonly Goverment _instance = new Goverment();

		// Make constructor private
		private Goverment()
		{
			Console.WriteLine("Constructor!");
		}

		// Method to get the instance
		public static Goverment Instance
		{
			get { return _instance; }
		}
	}
}
